mailpoet/lib/Mailer/MailerFactory.php

```diff
@@ -102,9 +102,7 @@ public function buildMailer(array $mailerConfig = null, array $sender = null, ar
       default:
         throw new InvalidStateException(__('Mailing method does not exist.', 'mailpoet'));
     }
-    $mailer = new Mailer();
-    $mailer->mailerInstance = $mailerInstance;
-    return $mailer;
+    return new Mailer($mailerInstance);
```

Comment:
Also here, I'd suggest renaming of `$mailerInstance` to `$mailerMethod` or just `$method`.